vic/extensions/rout_rvic/src/rout_alloc.c

```diff
@@ -0,0 +1,126 @@
+        log_err(
+            "Memory allocation error in rout.rout_param.source_time_offset().");
+    }
+    rout.rout_param.source_x_ind = (int *) malloc(
+        rout.rout_param.nSources * sizeof(int));
+    if (rout.rout_param.source_x_ind == NULL) {
+        log_err("Memory allocation error in rout.rout_param.source_x_ind().");
+    }
+    rout.rout_param.source_y_ind = (int *) malloc(
+        rout.rout_param.nSources * sizeof(int));
+    if (rout.rout_param.source_y_ind == NULL) {
+        log_err("Memory allocation error in rout.rout_param.source_y_ind().");
+    }
+    rout.rout_param.source_lat = (double *) malloc(
+        rout.rout_param.nSources * sizeof(double));
+    if (rout.rout_param.source_lat == NULL) {
+        log_err("Memory allocation error in rout.rout_param.source_lat().");
+    }
+    rout.rout_param.source_lon = (double *) malloc(
+        rout.rout_param.nSources * sizeof(double));
+    if (rout.rout_param.source_lon == NULL) {
+        log_err("Memory allocation error in rout.rout_param.source_lon().");
+    }
+    rout.rout_param.source_VIC_index = (int *) malloc(
+        rout.rout_param.nSources * sizeof(int));
+    if (rout.rout_param.source_VIC_index == NULL) {
+        log_err(
+            "Memory allocation error in rout.rout_param.source_VIC_index().");
+    }
+    rout.rout_param.outlet_lat = (double *) malloc(
+        rout.rout_param.nOutlets * sizeof(double));
+    if (rout.rout_param.outlet_lat == NULL) {
+        log_err("Memory allocation error in rout.rout_param.outlet_lat().");
+    }
+    rout.rout_param.outlet_lon = (double *) malloc(
+        rout.rout_param.nOutlets * sizeof(double));
+    if (rout.rout_param.outlet_lon == NULL) {
+        log_err("Memory allocation error in rout.rout_param.outlet_lon().");
+    }
+    rout.rout_param.outlet_VIC_index = (int *) malloc(
+        rout.rout_param.nOutlets * sizeof(int));
+    if (rout.rout_param.outlet_VIC_index == NULL) {
+        log_err(
+            "Memory allocation error in rout.rout_param.outlet_VIC_index().");
+    }
+    rout.rout_param.unit_hydrograph = (double *) malloc(
+        rout.rout_param.nSources * rout.rout_param.nTimesteps *
+        sizeof(double));
+    if (rout.rout_param.unit_hydrograph == NULL) {
+        log_err("Memory allocation error in rout.rout_param.unit_hydrograph().");
+    }
+    rout.rout_param.aggrunin =
+        (double *)malloc(local_domain.ncells_active * sizeof(double));
+    if (rout.rout_param.aggrunin == NULL) {
+        log_err("Memory allocation error in rout.rout_param.aggrunin().");
+    }
+    rout.discharge = (double *)malloc(local_domain.ncells_active * sizeof(double));
+    if (rout.discharge == NULL) {
+        log_err("Memory allocation error in rout.rout_param.discharge().");
+    }
+
+    // Allocate memory for the ring
+    rout.ring = (double *)malloc(
+        rout.rout_param.nTimesteps * rout.rout_param.nOutlets *
+        sizeof(double));
+    if (rout.ring == NULL) {
```

Comment:
Indeed you are right. I changed it for the ring you mentioned but also for all the other mallocs in this file.
However I think it is not nescessary to add extra brackets `sizeof(*(rout.ring)));` could also be `sizeof(*rout.ring));` is'nt it?